spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -844,10 +844,16 @@ def interpreter_versions(self):
             import subprocess
             versions = {}
             pyexec = CONF.get('main_interpreter', 'executable')
-            py_cmd = '%s -c "import sys; print(sys.version)"' % pyexec
-            ipy_cmd = ('%s -c "import IPython.core.release as r; print(r.version)"'
-                       % pyexec)
+            py_cmd = u'%s -c "import sys; print(sys.version)"' % pyexec
+            ipy_cmd = (
+                u'%s -c "import IPython.core.release as r; print(r.version)"'
+                % pyexec
+            )
             for cmd in [py_cmd, ipy_cmd]:
+                # Testing
+                if PY2:

```

Comment:
If this is necessary only for testing, then please remove the comment above and change this to

    if running_under_pytest() and PY2:

Reply:
I think this is a bug that we were not checking. What do you think?

Reply:
Probably. If that's the case, then please remove the `# Testing` comment in the line above (that's what confused me in the first place).

Reply:
Fixed!